aten/src/ATen/core/type.cpp

```diff
@@ -1661,8 +1657,12 @@ ClassType::ClassType(
       doc_string_(std::move(doc_string)),
       unresolved_class_attributes_(std::move(unresolved_class_attributes)) {}
 
-const std::vector<torch::jit::Function*>& ClassType::methods() const {
-  return methods_;
+const std::vector<torch::jit::Function*> ClassType::methods() const {
+  std::vector<torch::jit::Function*> methods;
```

Comment:
I know it is important to keep the signature of the method unchanged for backwards compatibility, but now the vector has to be created anew and returned each time the method is called.

Do you know if this cancels out the benefit of O(1) lookups?